tests/functests/instancetype_test.go

```diff
@@ -0,0 +1,137 @@
+				vm.Spec.Preference = &v1.PreferenceMatcher{
+					Name: preference.Name,
+				}
+
+				_, err := virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError(ContainSubstring(
+					"admission webhook \"virtualmachine-validator.kubevirt.io\" denied the request: "+
+						"failure checking preference requirements: "+
+						"insufficient Memory resources of 64M provided by instance type, preference requires %s",
+					preference.Spec.Requirements.Memory.Guest.String(),
+				)))
+			}
+			Expect(virtClient.VirtualMachineClusterInstancetype().Delete(context.Background(), "tiny-instancetype", metav1.DeleteOptions{})).ToNot(HaveOccurred())
+		})
+
+		It("can be created when enough resources are provided", func() {
+			vm := randomVMWithInstancetype("u1.large")
+
+			clusterPreferences, err := virtClient.VirtualMachineClusterPreference().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(clusterPreferences.Items).ToNot(BeEmpty())
+
+			for _, preference := range clusterPreferences.Items {
+				vm.Name = fmt.Sprintf("test-vm-%s", preference.Name)
+				vm.Spec.Preference = &v1.PreferenceMatcher{
+					Name: preference.Name,
+				}
+
+				vm, err := virtClient.VirtualMachine(testNamespace).Create(context.Background(), vm)
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
+					vm, err := virtClient.VirtualMachine(testNamespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					if !vm.Status.Ready {
+						return fmt.Errorf("vm not ready")
+					}
+					return nil
+				}, 60*time.Second, time.Second).Should(Succeed())
+
+				err = virtClient.VirtualMachine(testNamespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
+	})
+})
+
+func randomVMWithInstancetype(instancetypeName string) *v1.VirtualMachine {
+	name := "test-vm-" + rand.String(5)
+	vmLabels := map[string]string{"name": name}
+
+	vm := &v1.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: testNamespace,
+		},
+		Spec: v1.VirtualMachineSpec{
+			Running: pointer.Bool(true),
+			Instancetype: &v1.InstancetypeMatcher{
+				Name: instancetypeName,
+			},
+			Template: &v1.VirtualMachineInstanceTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:    vmLabels,
+					Namespace: testNamespace,
+				},
+				Spec: v1.VirtualMachineInstanceSpec{
+					Volumes: []v1.Volume{
+						{
+							Name: "disk",
+							VolumeSource: v1.VolumeSource{
+								ContainerDisk: &v1.ContainerDiskSource{
+									Image: "quay.io/containerdisks/fedora:latest",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	vm.SetGroupVersionKind(schema.GroupVersionKind{Group: v1.GroupVersion.Group, Kind: "VirtualMachine", Version: v1.GroupVersion.Version})
```

Comment:
Can you do this in the `vm` definition above?

Reply:
Also, is it really required?

Reply:
No, I will drop it